x/cosmwasmpool/model/codec.go

```diff
@@ -0,0 +1,55 @@
+package model
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	authzcodec "github.com/cosmos/cosmos-sdk/x/authz/codec"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
+
+	"github.com/osmosis-labs/osmosis/v15/x/cosmwasmpool/types"
+	poolmanagertypes "github.com/osmosis-labs/osmosis/v15/x/poolmanager/types"
+)
+
+func RegisterCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterConcrete(&CosmWasmPool{}, "osmosis/cosmwasm-pool", nil)
+	cdc.RegisterConcrete(&MsgCreateCosmWasmPool{}, "osmosis/cosmwasm-create-pool", nil)
+}
+
+func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
+	registry.RegisterInterface(
+		"osmosis.poolmanager.v1beta1.PoolI",
+		(*poolmanagertypes.PoolI)(nil),
+		&CosmWasmPool{},
+	)
+	registry.RegisterInterface(
+		"osmosis.cosmwasmpool.v1beta1.CosmWasmExtension",
+		(*types.CosmWasmExtension)(nil),
+		&CosmWasmPool{},
```

Comment:
This code block is the reason why [fdc16a4](https://github.com/osmosis-labs/osmosis/pull/4675/commits/fdc16a4e6948a88731ebf657e0c48fa8bc99ebba) is needed.

I'm open to suggestions on how to reduce the need for the boilerplate in [fdc16a4](https://github.com/osmosis-labs/osmosis/pull/4675/commits/fdc16a4e6948a88731ebf657e0c48fa8bc99ebba)

The way this works is that there is a proto-generated `CosmWasmPool` model that is written to state. It only implements `PoolI` and `CosmWasmExtensions`  for registering interfaces.
 
There is also a Go struct wrapper [` Pool`](https://github.com/osmosis-labs/osmosis/blob/fdc16a4e6948a88731ebf657e0c48fa8bc99ebba/x/cosmwasmpool/model/pool.go#L15-L18). It is not proto-generated because it must contain `WasmKeeper` for interacting with the contract. As a result, I'm unable to `RegisterInterface`s for `Pool` and have to rely on the verbose hack in [fdc16a4](https://github.com/osmosis-labs/osmosis/pull/4675/commits/fdc16a4e6948a88731ebf657e0c48fa8bc99ebba)